### PR TITLE
[idrac_user] Role

### PR DESCRIPTION
**Pull request by Kritika-Bhateja-03**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: .ansible-lint-ignore [hunk 1]
```diff
@@ -121,3 +121,5 @@ roles/idrac_attributes/molecule/default/converge.yml var-naming[no-role-prefix]
 roles/idrac_attributes/molecule/idrac_attr/converge.yml var-naming[no-role-prefix]
 roles/idrac_attributes/molecule/lifecycle_controller_attr/converge.yml var-naming[no-role-prefix]
 roles/idrac_attributes/molecule/system_attr/converge.yml var-naming[no-role-prefix]
+
```
Comment (sachin-apa) [inline .ansible-lint-ignore:128]: Any reason for this new line?

Comment (Kritika-Bhateja-03) [reply .ansible-lint-ignore:128]: As it a new role, separated it. I saw a pattern followed that each role/module is separated by a blank line.

File: roles/idrac_user/vars/main.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+# vars file for idrac_user
+idrac_user_delegate:
+  "{{ lookup('ansible.builtin.env', 'RUNON', default='localhost') }}"
+idrac_user_uri_headers:
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_user/vars/main.yml:5]: If we have used uri_module in only one place through out the role, I feel we should not populate vars/main.yml with these values, we can simply pass these value directly there.

Comment (Kritika-Bhateja-03) [reply roles/idrac_user/vars/main.yml:5]: Lets use vars/main.yml itself as its specifically for vars. Also we might enhance few things in module and role later.

File: roles/idrac_user/README.md [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+<thead>
+  <tr>
+    <th>Name</th>
+    <th>Sample</th>
+    <th>Description</th>
+  </tr>
+</thead>
+  <tbody>
+    <tr>
+      <td>idrac_user_out</td>
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_user/README.md:227]: Please update proper return msg, which is coming from modules, idrac_user_out will return module output, I can see status and error_info is missing

https://github.com/dell/dellemc-openmanage-ansible-modules/blob/collections/plugins/modules/idrac_user.py#L159

File: roles/idrac_user/tasks/get_user.yml
Comment (ABHISHEK-SINHA10) [inline roles/idrac_user/tasks/get_user.yml:1]: Any reason why we have used uri module and not our own module idrac_user_info, 

- name: Fetching user info
  dellemc.openmanage.idrac_user_info:
    idrac_ip: "{{ lookup('env', 'IDRAC_IP') }}"
    idrac_user: "{{ lookup('env', 'IDRAC_USER') }}"
    idrac_password: "{{ lookup('env', 'IDRAC_PASSWORD') }}"
    validate_certs: false
    username: "{{ idrac_user_name }}"
  
  
  Something like this ?

Comment (ABHISHEK-SINHA10) [reply roles/idrac_user/tasks/get_user.yml:1]: resolving this, will raise CR for this.

File: roles/idrac_user/README.md (hunk 3 above)
Comment (ABHISHEK-SINHA10) [reply roles/idrac_user/README.md:227]: resolving it as status doesn't add value

**Review by ABHISHEK-SINHA10 — DISMISSED** [review]

**Review by sachin-apa — DISMISSED** [review]

File: roles/idrac_user/tasks/get_user.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+---
+- name: Fetch user accounts
+  ansible.builtin.uri:
+    url: "https://{{ hostname }}:{{ https_port }}/redfish/\
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_user/tasks/get_user.yml]: pass ipwrap filter for hostname, otherwise will fail for ipv6

Comment (Kritika-Bhateja-03) [reply roles/idrac_user/tasks/get_user.yml]: I had went through the module, it was no where mentioned that the module supports IPv6. So didn't use.

File: roles/idrac_user/tasks/get_user.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+---
+- name: Set user_name based on new_user_name
+  ansible.builtin.set_fact:
+    user_name: "{{ new_user_name if new_user_name is defined else user_name }}"
```
Comment (sachin-apa) [inline roles/idrac_user/tasks/get_user.yml]: change this fact name to use the format idrac_user_present_name

Comment (Kritika-Bhateja-03) [reply roles/idrac_user/tasks/get_user.yml]: done

**Review by ABHISHEK-SINHA10 — APPROVED** [review]

**Review by ShivamSh3 — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Pull request merged**